assets/toolbar.js

```diff
@@ -1,42 +1,156 @@
-    };
+            };
+            xhr.send(settings.data || '');
+        },
+        url,
+        div,
+        toolbarEl = findToolbar(),
+        barSelector = '.yii-debug-toolbar__bar',
+        viewSelector = '.yii-debug-toolbar__view',
+        blockSelector = '.yii-debug-toolbar__block',
+        togglerSelector = '.yii-debug-toolbar__toggler',
+        minimizerSelector = '.yii-debug-toolbar__minimizer',
+
+        CACHE_KEY = 'yii-debug-toolbar',
+        STATE_MAXIMIZED = 'maximized',
+
+        maximizedClass = 'yii-debug-toolbar_minimized',
+        viewActiveClass = 'yii-debug-toolbar_view_active',
+        blockClass = 'yii-debug-toolbar__block',
+        blockActiveClass = 'yii-debug-toolbar__block_active';
+
+    if (toolbarEl) {
+        toolbarEl.style.display = 'block';
+        url = toolbarEl.getAttribute('data-url');
 
-    var e = document.getElementById('yii-debug-toolbar');
-    if (e) {
-        e.style.display = 'block';
-        var url = e.getAttribute('data-url');
         ajax(url, {
             success: function (xhr) {
-                var div = document.createElement('div');
+                div = document.createElement('div');
                 div.innerHTML = xhr.responseText;
-                e.parentNode.replaceChild(div, e);
-                if (window.localStorage) {
-                    var pref = localStorage.getItem('yii-debug-toolbar');
-                    if (pref == 'minimized') {
-                        document.getElementById('yii-debug-toolbar').style.display = 'none';
-                        document.getElementById('yii-debug-toolbar-min').style.display = 'block';
-                    }
-                }
+
+                toolbarEl.parentNode.replaceChild(div, toolbarEl);
+
+                showToolbar();
             },
             error: function (xhr) {
-                e.innerHTML = xhr.responseText;
+                toolbarEl.innerHTML = xhr.responseText;
             }
         });
     }
+
+    function showToolbar() {
+        var el = findToolbar(),
+            barEl = el.querySelector(barSelector),
+            blockEls = barEl.querySelectorAll(blockSelector),
+            viewEl = el.querySelector(viewSelector),
+            iframeEl = viewEl.querySelector('iframe'),
+            toggleEl = el.querySelector(togglerSelector),
+            minimizeEl = el.querySelector(minimizerSelector),
+            viewHeight = function () {
+                return (window.innerHeight * 0.7) + 'px';
+            },
+            showView = function () {
+                el.classList.add(viewActiveClass);
+                viewEl.style.height = viewHeight();
+            },
+            hideView = function () {
+                findToolbar().classList.remove(viewActiveClass);
+                viewEl.style.height = '';
+                removeActiveBlocksCls();
+            },
+            removeActiveBlocksCls = function () {
+                [].forEach.call(blockEls, function (el) {
```

Comment:
`querySelectorAll` returns `NodeList` object, that implements array-like interface, but does not have `Array` methods, so we can iterate over list by calling `Array.prototype.forEach`, this behavior similar with `arguments` pseudo array inside js function.